### DISCUSSION TITLE
NullIf Function

### DISCUSSION THREAD

**Discussion by jasonkres** [datalust/seq-tickets#1453] [Ideas]
This is an SQL standard function. It is often used together with `Coalesce`. Without it, some expressions (e.g., that would be common in data cleansing) are very awkward and lengthy.

**Example 1**

Instead of:

```sql
if subexpression = '' then null else subexpression
```

Can write:

```sql
NullIf(subexpression, '')
```

**Example 2**

Instead of:

```sql
if subexpression is null or subexpression = '' then 'unavailable' else subexpression
```

Can write:
```
Coalesce(NullIf(subexpression, ''), 'unavailable')
```

The advantage of this becomes clearer when noting that the subexpression is repeated multiple times in the `if`/`then`/`else` but only once in the `Coalesce`/`NullIf` equivalent. The subexpression could be very lengthy.

**Comment by nblumhardt** [comment]
Thanks for the suggestion! We'll give this some thought 👍 

`let` can be used in Seq today, to eliminate repeated subexpressions:

```sql
let |s: subexpression| if s is null or s = '' then 'unavailable' else s
```

**Comment by nblumhardt** [comment]
Thanks again for this suggestion, @jasonkres . We're moving it to a discussion for now, as we don't expect it will make the 2022.1 or .2 milestones, but we're keen to keep the thread alive.
